test/spawn.jl

```diff
@@ -173,14 +173,17 @@ exename = joinpath(JULIA_HOME, Base.julia_exename())
 if valgrind_off
     # If --trace-children=yes is passed to valgrind, we will get a
     # valgrind banner here, not "Hello World\n".
-    @test readall(pipe(`$exename -f -e 'println(STDERR,"Hello World")'`, stderr=`cat`)) == "Hello World\n"
+    @test readall(pipeline(`$exename -f -e 'println(STDERR,"Hello World")'`, stderr=`cat`)) == "Hello World\n"
+    p = Pipe()
+    run(pipeline(`$exename -f -e 'println(STDERR,"Hello World")'`, stderr = p))
```

Comment:
I think this is similar to what I proposed at some point – the idea was that a `Pipe` object would be similar to a `cat` process but without any process, just going through the kernel.